File: collect_app/src/main/java/org/odk/collect/android/network/ConnectivityProvider.java
package org.odk.collect.android.network;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

import org.odk.collect.android.application.Collect;

public class ConnectivityProvider implements NetworkStateProvider {

    public boolean isDeviceOnline() {
        NetworkInfo networkInfo = getNetworkInfo();
        return networkInfo != null && networkInfo.isConnected();
    }

    public NetworkInfo getNetworkInfo() {
        return getConnectivityManager().getActiveNetworkInfo();
    }

    private ConnectivityManager getConnectivityManager() {
        return (ConnectivityManager) Collect.getInstance().getSystemService(Context.CONNECTIVITY_SERVICE);
    }
}

File: android-shared/src/main/java/org/odk/collect/androidshared/network/ConnectivityProvider.kt
package org.odk.collect.androidshared.network

import android.content.Context
import android.net.ConnectivityManager
import android.net.NetworkInfo

class ConnectivityProvider(private val context: Context) : NetworkStateProvider {
    override val isDeviceOnline: Boolean
        get() {
            val networkInfo = networkInfo
            return networkInfo != null && networkInfo.isConnected
        }

    override val networkInfo: NetworkInfo?
        get() = connectivityManager.activeNetworkInfo

    private val connectivityManager: ConnectivityManager
        get() = context.getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager
}
